File: backend/src/main/java/com/primetrade/backend/service/ProductService.java
package com.primetrade.backend.service;

import java.util.List;
import org.springframework.data.domain.Page;
import com.primetrade.backend.dto.ProductRequest;
import com.primetrade.backend.dto.ProductResponse;

public interface ProductService {

    ProductResponse createProduct(ProductRequest request);

    List<ProductResponse> getAllProducts();

    ProductResponse getProductById(Long id);

    ProductResponse updateProduct(Long id, ProductRequest request);

    String deleteProduct(Long id);
    
    List<ProductResponse> searchProducts(String keyword);
    
    Page<ProductResponse> getProducts(
            int page,
            int size,
            String sortBy,
            String direction);
}

File: frontend/src/services/ProductService.js
import api from '../api/axiosConfig'

export const getProducts = ({ page, size, sortBy, direction }) =>
  api.get('/products/page', {
    params: {
      page,
      size,
      sortBy,
      direction,
    },
  })

export const createProduct = (product) => api.post('/products', product)
export const updateProduct = (id, product) => api.put(`/products/${id}`, product)
export const deleteProduct = (id) => api.delete(`/products/${id}`)
export const searchProducts = (query) =>
  api.get('/products/search', {
    params: {
      keyword: query,
    },
  })